src/GraphRequest.ts

```diff
@@ -244,9 +236,105 @@ export class GraphRequest {
+	 * @param {string|KeyValuePairObjectStringNumber} queryDictionaryOrString - The query parameter
+	 * @returns The same GraphRequest instance that is being called with
+	 */
+	private parseQueryParameter(queryDictionaryOrString: string | KeyValuePairObjectStringNumber): GraphRequest {
+		if (typeof queryDictionaryOrString === "string") {
+			if (queryDictionaryOrString.charAt(0) === "?") {
+				queryDictionaryOrString = queryDictionaryOrString.substring(1, queryDictionaryOrString.length);
+			}
+
+			if (queryDictionaryOrString.indexOf("&") !== -1) {
+				const queryParams = queryDictionaryOrString.split("&");
+				for (const str of queryParams) {
+					this.parseQueryParamenterString(str);
+				}
+			} else {
+				this.parseQueryParamenterString(queryDictionaryOrString);
+			}
+		} else if (queryDictionaryOrString.constructor === Object) {
+			for (const key in queryDictionaryOrString) {
+				if (queryDictionaryOrString.hasOwnProperty(key)) {
+					this.setURLComponentsQueryParamater(key, queryDictionaryOrString[key]);
+				}
+			}
+		} else {
+			/*Push values which are not of key-value structure. 
+			Example-> Handle an invalid input->.query(123) and let the Graph API respond with the error in the URL*/ this.urlComponents.otherURLQueryOptions.push(queryDictionaryOrString);
+		}
+
+		return this;
+	}
+
+	/**
+	 * @private
+	 * Parses the query parameter of string type to set the urlComponents property of the GraphRequest object
+	 * @param {string} queryParameter - the query parameters
+	 * returns nothing
+	 */
+	private parseQueryParamenterString(queryParameter: string): void {
+		/* The query key-value pair must be split on the first equals sign to avoid errors in parsing nested query parameters.
+				 Example-> "/me?$expand=home($select=city)" */
+		if (this.isValidQueryKeyValuePair(queryParameter)) {
+			const indexOfFirstEquals = queryParameter.indexOf("=");
+			const paramKey = queryParameter.substring(0, indexOfFirstEquals);
+			const paramValue = queryParameter.substring(indexOfFirstEquals + 1, queryParameter.length);
+			this.setURLComponentsQueryParamater(paramKey, paramValue);
+		} else {
+			/* Push values which are not of key-value structure. 
+			Example-> Handle an invalid input->.query(test), .query($select($select=name)) and let the Graph API respond with the error in the URL*/
+			this.urlComponents.otherURLQueryOptions.push(queryParameter);
+		}
+	}
+
+	/**
+	 * @private
+	 * Sets values into the urlComponents property of GraphRequest object.
+	 * @param {string} paramKey - the query parameter key
+	 * @param {string} paramValue - the query paramter value
+	 * @returns nothing
+	 */
+	private setURLComponentsQueryParamater(paramKey: string, paramValue: string | number): void {
+		if (oDataQueryNames.indexOf(paramKey) !== -1) {
+			const currentValue = this.urlComponents.oDataQueryParams[paramKey];
+			const isValueAppendable = currentValue && (paramKey === "$expand" || paramKey === "$select" || paramKey === "$orderby");
```

Comment:
Then it gets parsed as a normal query parameter.

Reply:
which I don't think is the right thing. if I, as a consumer, pass "select" or "$Select" the service will understand those things. So our client should not be limiting when the service supports it.

Reply:
if the user passes example '?select=xyz', then the url built is '?select=xyz'.
if the user passes  example '?**$**select=xyz', then the url built is '?**$**select=xyz'.

The query parameters passed by the user is passed as inputted to the service. And the service will validate the url. The client will not be validating the url .

Reply:
I'm not sure I'm following this response? if paramKey is "select" it'll derail the isValueAppendable logic and following.

Reply:
select != $select is not valid. I tried the following on graph explorer.  
https://graph.microsoft.com/v1.0/me/?$select=displayname -> this works
https://graph.microsoft.com/v1.0/me/?select=displayname->  Error -> Unrecognized query argument specified 'select'. 

The following are test cases which work in Graph and example how the input will be constructing back the URL -> 
"me?$select=displayName": "https://graph.microsoft.com/v1.0/me?$select=displayName",
"me?select=displayName": "https://graph.microsoft.com/v1.0/me?select=displayName".

If the user inputs 'select' instead of '$select' then the client will not be correcting this. The user will get an error from the service.

Reply:
yet this works https://graph.microsoft.com/v1.0/me/drive?select=description

FYI the directory services are still operating with oDdata v3 and AGS tries to do some "translation" from odata v4,  but it's far from perfect. The bottom line being, you want to "test" odata stuff both on identity endpoints and on non-identity endpoints.

Reply:
FYI, see the "tip" https://docs.microsoft.com/en-us/graph/query-parameters

Reply:
I think it will be only safe to assume that select and $select are the same, if it is consistently seen in service. Example- 
It is consistent if https://graph.microsoft.com/v1.0/me/?select=displayname  works as well . 
For now, the client should parse select and $select as two different keywords. The client will work correctly if the user inputs $select as expected. 
The current PR contains the changes to fix the parsing logic for nested query parameter values. It is not changing how to the odata-query parameter keys are handled currently.

Reply:
Created this issue #341.